### PR TITLE
Adjust documentation to Bazel and Tectonic cli

### PR DESCRIPTION
**Pull request by mxinden**
- Extend `### Hacking` session of root README.md
- Remove obsolete contrib/tectonic-cli-examples/README.md

//CC @spangenberg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,65 +35,41 @@ These instructions can be used for the official stable platforms listed above, a
...
-```
-PLATFORM=azure CLUSTER=my-cluster make apply
+6. Install Tectonic cluster
+```shell
+tectonic install --dir=<CLUSTER_NAME>
 ```
 
+7. Teardown Tectonic cluster
+```shell
+tectonic delete <CLUSTER_NAME>
```
Comment (enxebre) [inline README.md]: is it delete now, or still destroy?

Comment (mxinden) [reply README.md]: Thanks. Fixed.

File: README.md [hunk 2]
```diff
@@ -35,65 +35,55 @@ These instructions can be used for the official stable platforms listed above, a
...
+    ```
+
+7. Install Tectonic cluster
+    ```shell
+    tectonic install --dir=$CLUSTER_NAME
+    ```
+
+8. Teardown Tectonic cluster
+    ```shell
+    tectonic destroy $CLUSTER_NAME
```
Comment (squat) [inline README.md]: Is this correct? Install takes a `—dir` flag but destroy does not? If so, this is pretty weird UX and warrants a change in the CLI. Surely both the flag and the cluster name argument serve the same purpose and can fulfill the same needs

**Comment by mxinden** [comment]
@squat @enxebre can you give this another review?

File: README.md (hunk 2 above)
Comment (squat) [reply README.md]: thanks for filing that. we really need to polish up the CLI

**Pull request merged**